internal/controller/ols_app_server_assets.go

```diff
@@ -116,37 +116,33 @@ func (r *OLSConfigReconciler) generateOLSConfigMap(cr *olsv1alpha1.OLSConfig) (*
 		}
 		providerConfigs = append(providerConfigs, providerConfig)
 	}
-	// TODO: Update DB
-	// redisMaxMemory := intstr.FromString(RedisMaxMemory)
-	// redisMaxMemoryPolicy := RedisMaxMemoryPolicy
-	// redisSecretName := RedisSecretName
-	// redisConfig := cr.Spec.OLSConfig.ConversationCache.Redis
-	// if redisConfig.MaxMemory != nil && redisConfig.MaxMemory.String() != "" {
-	// 	redisMaxMemory = *cr.Spec.OLSConfig.ConversationCache.Redis.MaxMemory
-	// }
-	// if redisConfig.MaxMemoryPolicy != "" {
-	// 	redisMaxMemoryPolicy = cr.Spec.OLSConfig.ConversationCache.Redis.MaxMemoryPolicy
-	// }
-	// if redisConfig.CredentialsSecret != "" {
-	// 	redisSecretName = cr.Spec.OLSConfig.ConversationCache.Redis.CredentialsSecret
-	// }
-	// redisPasswordPath := path.Join(CredentialsMountRoot, redisSecretName, OLSComponentPasswordFileName)
-	// conversationCache := ConversationCacheConfig{
-	// 	Type: string(OLSDefaultCacheType),
-	// 	Redis: RedisCacheConfig{
-	// 		Host:            strings.Join([]string{RedisServiceName, r.Options.Namespace, "svc"}, "."),
-	// 		Port:            RedisServicePort,
-	// 		MaxMemory:       &redisMaxMemory,
-	// 		MaxMemoryPolicy: redisMaxMemoryPolicy,
-	// 		PasswordPath:    redisPasswordPath,
-	// 		CACertPath:      path.Join(OLSAppCertsMountRoot, RedisCertsSecretName, RedisCAVolume, "service-ca.crt"),
-	// 	},
-	// }
 
+	postgresSharedBuffers := intstr.FromString(PostgresSharedBuffers)
+	postgresMaxConnections := PostgresMaxConnections
+	postgresSecretName := PostgresSecretName
+	postgresConfig := cr.Spec.OLSConfig.ConversationCache.Postgres
+	if postgresConfig.SharedBuffers != nil && postgresConfig.SharedBuffers.String() != "" {
+		postgresSharedBuffers = *cr.Spec.OLSConfig.ConversationCache.Postgres.SharedBuffers
+	}
+	if postgresConfig.MaxConnections != (-1) {
+		postgresMaxConnections = cr.Spec.OLSConfig.ConversationCache.Postgres.MaxConnections
+	}
+	if postgresConfig.CredentialsSecret != "" {
+		postgresSecretName = cr.Spec.OLSConfig.ConversationCache.Postgres.CredentialsSecret
+	}
+	postgresPasswordPath := path.Join(CredentialsMountRoot, postgresSecretName, OLSComponentPasswordFileName)
 	conversationCache := ConversationCacheConfig{
-		Type: "memory",
-		Memory: MemoryCacheConfig{
-			MaxEntries: 1000,
+		Type: string(OLSDefaultCacheType),
+		Postgres: PostgresCacheConfig{
+			Host:           strings.Join([]string{PostgresServiceName, r.Options.Namespace, "svc"}, "."),
+			Port:           PostgresServicePort,
+			User:           PostgresDefaultUser,
+			DbName:         PostgresDefaultDbName,
+			SharedBuffers:  &postgresSharedBuffers,
```

Comment:
does OLS actually need this value?  it seems like something only the operator needs (to configure the postgres deployment), not something the ols service needs to know about, unless it impacts the ols postgres client configuration?

possibly the same for the maxconnections (if it doesn't impact the client, ols config yaml doesn't need it, right?  just the operator)

Reply:
Thanks for catching this. Removed it from the OLS config as it is not required.